### PR TITLE
Rainbow test

### PR DESCRIPTION
**Pull request by zach-planet-nine**
This adds the rainbow test to the node cli (I'd like to update the other clis too, but it's lower priority than getting the rest of the example servers working). Right now it works with the javascript, typescript, rust, and c# implementations. The Java and Kotlin servers need to be updated to conform to the naming we settled on, and we need to build a python server.

### CODE REVIEW

File: src/rust/examples/slapper/src/requests/register.rs
Comment (FssAy) [inline src/rust/examples/slapper/src/requests/register.rs:1]: Rust uses `snake_case` naming convention, but Sessionless uses `camelCase`. In order to comply with both it's possible to use serde attributes. These attributes can help rename the struct's fields when serialized.

The easiest way to fix that would be adding:
```rust
#[serde(rename_all="camelCase")]
```
on the line `32` just above:
```rust
struct RegisterPayload<'a> {
```

Then we can use `snake_case` variable names as all of them will be transformed to `camelCase` automatically.

**Review by FssAy — CHANGES_REQUESTED** [review]

File: src/rust/examples/slapper/src/requests/register.rs
Comment (zach-planet-nine) [reply src/rust/examples/slapper/src/requests/register.rs:1]: Woof, this change must have been from long ago and I committed it and forgot about it. I can either revert to what's on main or make the change you suggest and clean up anything that might break.

Comment (FssAy) [reply src/rust/examples/slapper/src/requests/register.rs:1]: I mean, if you revert that would create a bug as the fields won't be serialized in `camelCase`.

Comment (zach-planet-nine) [reply src/rust/examples/slapper/src/requests/register.rs:1]: well that answers that. How's this update look?

Comment (FssAy) [reply src/rust/examples/slapper/src/requests/register.rs:1]: Top 👌

File: src/java/spring/src/main/java/app/planentnine/springsessionless/adapter/web/UserController.java
Comment (FssAy) [inline src/java/spring/src/main/java/app/planentnine/springsessionless/adapter/web/UserController.java:1]: Is it part of the Java implementation? I guess there is no longer need for the debug comments haha

File: src/csharp/examples/server/obj/project.nuget.cache
Comment (FssAy) [inline src/csharp/examples/server/obj/project.nuget.cache:1]: From what I've read just now, this file shouldn't be pushed to source control and added to `.gitignore`.

File: src/csharp/examples/server/users.json
Comment (FssAy) [inline src/csharp/examples/server/users.json:1]: I think we could also add this file to `.gitignore`, I believe it's runtime generated and will be overwritten every time.

File: src/java/spring/src/main/java/app/planentnine/springsessionless/adapter/web/.MessageController.java.swp
Comment (FssAy) [inline src/java/spring/src/main/java/app/planentnine/springsessionless/adapter/web/.MessageController.java.swp:1]: I think this also got here by an accident. I've found out it's a temporary file generated by Vim, but I am not sure.

**Review by FssAy — CHANGES_REQUESTED** [review]
I've just gazed through the code to find any parts that might got pushed by an accident.

File: src/java/spring/src/main/java/app/planentnine/springsessionless/adapter/web/.MessageController.java.swp
Comment (zach-planet-nine) [reply src/java/spring/src/main/java/app/planentnine/springsessionless/adapter/web/.MessageController.java.swp:1]: yeah... man I don't know what I was doing two weeks ago.

File: src/java/spring/src/main/java/app/planentnine/springsessionless/adapter/web/UserController.java
Comment (zach-planet-nine) [reply src/java/spring/src/main/java/app/planentnine/springsessionless/adapter/web/UserController.java:1]: Ok I think this is ready for a second look. Sorry for the churn, and thanks for catching it.

File: src/csharp/examples/server/users.json
Comment (FssAy) [reply src/csharp/examples/server/users.json:1]: Great! I think that's the best solution here in that case.

**Review by FssAy — APPROVED** [review]
Looks good 👌

**Pull request merged**